opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1202,6 +1203,35 @@ namespace Opm
         }
     }
 
+    template <typename TypeTag>
+    double
+    WellInterface<TypeTag>::
+    wellIndex(const int perf, const IntensiveQuantities& intQuants, const SingleWellState& ws) const {
```

Comment:
This function does a lot of unnecessary work for the typical case of `D = 0`.  Is there no other way of structuring this work to avoid that complexity?